Logback format is not being recognized anymore

On startup the following error can be see:

```
05-Oct-2022 16:38:36.299 SEVERE [main] org.apache.catalina.startup.HostConfig.deployDirectory Error deploying web application directory [/usr/local/tomcat/webapps/Biblivre6]
	java.lang.IllegalStateException: Error starting child
		at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:729)
		at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:698)
		at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:747)
		at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1188)
		at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:2023)
		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
		at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
		at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:123)
		at org.apache.catalina.startup.HostConfig.deployDirectories(HostConfig.java:1098)
		at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:480)
		at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1708)
		at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:320)
		at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
		at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423)
		at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:366)
		at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:946)
		at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:886)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
		at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1396)
		at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1386)
		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
		at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
		at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:145)
		at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:919)
		at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:265)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
		at org.apache.catalina.core.StandardService.startInternal(StandardService.java:432)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
		at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:930)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
		at org.apache.catalina.startup.Catalina.start(Catalina.java:795)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:568)
		at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:345)
		at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:476)
	Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Catalina].StandardHost[localhost].StandardContext[/Biblivre6]]
		at org.apache.catalina.util.LifecycleBase.handleSubClassException(LifecycleBase.java:440)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:198)
		at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:726)
		... 37 more
	Caused by: java.lang.IllegalStateException: java.lang.IllegalStateException: Logback configuration error detected:
ERROR in ch.qos.logback.classic.pattern.ClassOfCallerConverter@3d1c7e7b - failed to parse integer string [1.] java.lang.NumberFormatException: For input string: "1."
ERROR in ch.qos.logback.classic.pattern.ClassOfCallerConverter@460f2b6c - failed to parse integer string [1.] java.lang.NumberFormatException: For input string: "1."
		at org.springframework.boot.context.logging.LoggingApplicationListener.initializeSystem(LoggingApplicationListener.java:347)
		at org.springframework.boot.context.logging.LoggingApplicationListener.initialize(LoggingApplicationListener.java:298)
		at org.springframework.boot.context.logging.LoggingApplicationListener.onApplicationEnvironmentPreparedEvent(LoggingApplicationListener.java:246)
		at org.springframework.boot.context.logging.LoggingApplicationListener.onApplicationEvent(LoggingApplicationListener.java:223)
		at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:176)
		at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:169)
		at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:143)
		at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:131)
		at org.springframework.boot.context.event.EventPublishingRunListener.multicastInitialEvent(EventPublishingRunListener.java:136)
		at org.springframework.boot.context.event.EventPublishingRunListener.environmentPrepared(EventPublishingRunListener.java:82)
		at org.springframework.boot.SpringApplicationRunListeners.lambda$environmentPrepared$2(SpringApplicationRunListeners.java:64)
		at java.base/java.lang.Iterable.forEach(Iterable.java:75)
		at org.springframework.boot.SpringApplicationRunListeners.doWithListeners(SpringApplicationRunListeners.java:118)
		at org.springframework.boot.SpringApplicationRunListeners.doWithListeners(SpringApplicationRunListeners.java:112)
		at org.springframework.boot.SpringApplicationRunListeners.environmentPrepared(SpringApplicationRunListeners.java:63)
		at org.springframework.boot.SpringApplication.prepareEnvironment(SpringApplication.java:357)
		at org.springframework.boot.SpringApplication.run(SpringApplication.java:307)
		at org.springframework.boot.web.servlet.support.SpringBootServletInitializer.run(SpringBootServletInitializer.java:174)
		at org.springframework.boot.web.servlet.support.SpringBootServletInitializer.createRootApplicationContext(SpringBootServletInitializer.java:154)
		at org.springframework.boot.web.servlet.support.SpringBootServletInitializer.onStartup(SpringBootServletInitializer.java:96)
		at org.springframework.web.SpringServletContainerInitializer.onStartup(SpringServletContainerInitializer.java:171)
		at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5168)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
		... 38 more
	Caused by: java.lang.IllegalStateException: Logback configuration error detected:
ERROR in ch.qos.logback.classic.pattern.ClassOfCallerConverter@3d1c7e7b - failed to parse integer string [1.] java.lang.NumberFormatException: For input string: "1."
ERROR in ch.qos.logback.classic.pattern.ClassOfCallerConverter@460f2b6c - failed to parse integer string [1.] java.lang.NumberFormatException: For input string: "1."
		at org.springframework.boot.logging.logback.LogbackLoggingSystem.loadConfiguration(LogbackLoggingSystem.java:234)
		at org.springframework.boot.logging.logback.LogbackLoggingSystem.reinitialize(LogbackLoggingSystem.java:290)
		at org.springframework.boot.logging.AbstractLoggingSystem.initializeWithConventions(AbstractLoggingSystem.java:73)
		at org.springframework.boot.logging.AbstractLoggingSystem.initialize(AbstractLoggingSystem.java:60)
		at org.springframework.boot.logging.logback.LogbackLoggingSystem.initialize(LogbackLoggingSystem.java:185)
		at org.springframework.boot.context.logging.LoggingApplicationListener.initializeSystem(LoggingApplicationListener.java:332)
		... 60 more
```

--- a/pom.xml
+++ b/pom.xml
@@ -316,6 +316,12 @@
 			<artifactId>snakeyaml</artifactId>
 			<version>1.32</version>
 		</dependency>
+
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-databind</artifactId>
+			<version>2.14.0-rc1</version>
+		</dependency>
 	</dependencies>
 
 	<build>

--- a/src/main/java/biblivre/core/ExtendedRequest.java
+++ b/src/main/java/biblivre/core/ExtendedRequest.java
@@ -54,17 +54,23 @@ public class ExtendedRequest extends HttpServletRequestWrapper {
     private Map<String, String> multiPartParameters;
     private Map<String, MemoryFile> multiPartFiles;
 
-    public ExtendedRequest(HttpServletRequest request) throws IOException, ServletException {
+    public ExtendedRequest(HttpServletRequest request, RequestParserHelper requestParserHelper)
+            throws IOException, ServletException {
         super(request);
 
-        String path = request.getServletPath();
+        String servletPath = request.getServletPath();
 
-        if (path.contains("static/")) {
+        if (servletPath.contains("static/")) {
             return;
         }
 
         this.loadMultiPart(request);
-        this.loadController();
+
+        String requestPath = this.getRequestURI().substring(this.getContextPath().length() + 1);
+
+        this.setController(
+                requestParserHelper.parseController(requestPath, getString("controller")));
+        this.setMustRedirectToSchema(requestParserHelper.isMustRedirectToSchema(requestPath));
         this.loadLanguage();
         this.loadTranslationsMap();
     }
@@ -288,27 +294,6 @@ public class ExtendedRequest extends HttpServletRequestWrapper {
                 }
             }
         }
-    }
-
-    private void loadController() {
-        String url = this.getRequestURI().substring(this.getContextPath().length() + 1);
-        String controller = null;
-
-        if (StringUtils.isNotBlank(url)) {
-            String[] urlArray = url.split("/");
-
-            if (urlArray.length > 1) {
-                controller = urlArray[1];
-            } else if (!url.endsWith("/")) {
-                this.setMustRedirectToSchema(true);
-            }
-        }
-
-        if (controller == null) {
-            controller = this.getString("controller");
-        }
-
-        this.setController(controller);
     }
 
     public void clearSessionAttributes(String schema) {

--- a/src/main/java/biblivre/core/RequestParserHelper.java
+++ b/src/main/java/biblivre/core/RequestParserHelper.java
@@ -1,0 +1,7 @@
+package biblivre.core;
+
+public interface RequestParserHelper {
+    boolean isMustRedirectToSchema(String path);
+
+    String parseController(String path, String defaultValue);
+}

--- a/src/main/java/biblivre/core/RequestParserHelperImpl.java
+++ b/src/main/java/biblivre/core/RequestParserHelperImpl.java
@@ -1,0 +1,40 @@
+package biblivre.core;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RequestParserHelperImpl implements RequestParserHelper {
+
+    @Override
+    public boolean isMustRedirectToSchema(String path) {
+        if (StringUtils.isNotBlank(path)) {
+            String[] urlArray = path.split("/");
+
+            if (urlArray.length <= 1 && !path.endsWith("/")) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    @Override
+    public String parseController(String path, String defaultValue) {
+        if (StringUtils.isNotBlank(path)) {
+            String[] urlArray = path.split("/");
+
+            String schema = urlArray[0];
+
+            if ("DigitalMediaController".equals(schema)) {
+                return "DigitalMediaController";
+            }
+
+            if (urlArray.length > 1) {
+                return urlArray[1];
+            }
+        }
+
+        return defaultValue;
+    }
+}

--- a/src/main/java/biblivre/core/controllers/ExtendedRequestResponseFilter.java
+++ b/src/main/java/biblivre/core/controllers/ExtendedRequestResponseFilter.java
@@ -21,6 +21,7 @@ package biblivre.core.controllers;
 
 import biblivre.core.ExtendedRequest;
 import biblivre.core.ExtendedResponse;
+import biblivre.core.RequestParserHelper;
 import biblivre.core.utils.Constants;
 import jakarta.servlet.Filter;
 import jakarta.servlet.FilterChain;
@@ -30,10 +31,12 @@ import jakarta.servlet.ServletResponse;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
 import java.io.IOException;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
 public class ExtendedRequestResponseFilter implements Filter {
+    @Autowired private RequestParserHelper requestParserHelper;
 
     @Override
     public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
@@ -49,7 +52,7 @@ public class ExtendedRequestResponseFilter implements Filter {
             // Avoid rewrapping if forwarding
             xRequest = (ExtendedRequest) request;
         } else {
-            xRequest = new ExtendedRequest((HttpServletRequest) request);
+            xRequest = new ExtendedRequest((HttpServletRequest) request, requestParserHelper);
         }
 
         if (response instanceof ExtendedResponse) {

--- a/src/main/java/biblivre/core/controllers/SchemaServlet.java
+++ b/src/main/java/biblivre/core/controllers/SchemaServlet.java
@@ -27,6 +27,7 @@ import biblivre.circulation.user.UserFields;
 import biblivre.core.ExtendedRequest;
 import biblivre.core.ExtendedResponse;
 import biblivre.core.IFCacheableJavascript;
+import biblivre.core.RequestParserHelper;
 import biblivre.core.SchemaThreadLocal;
 import biblivre.core.auth.AuthorizationPoints;
 import biblivre.core.configurations.Configurations;
@@ -203,13 +204,8 @@ public final class SchemaServlet extends HttpServlet {
             HttpServletRequest request, HttpServletResponse response, boolean headerOnly)
             throws ServletException, IOException {
         final String path = request.getServletPath();
-        final String realPath;
-
-        if (path.contains("static/")) {
-            realPath = path.substring(path.lastIndexOf("/static"));
-        } else {
-            realPath = path.substring(path.lastIndexOf("/extra"));
-        }
+        final String realPath =
+                path.substring(path.lastIndexOf(path.contains("static/") ? "/static" : "/extra"));
 
         if (realPath.endsWith(".i18n.js")
                 || realPath.endsWith(".form.js")
@@ -247,7 +243,7 @@ public final class SchemaServlet extends HttpServlet {
         RequestDispatcher rd = this.getServletContext().getNamedDispatcher("dispatcherServlet");
 
         ExtendedRequest wrapped =
-                new ExtendedRequest(request) {
+                new ExtendedRequest(request, getRequestParserHelper(request)) {
 
                     @Override
                     public String getServletPath() {
@@ -256,6 +252,15 @@ public final class SchemaServlet extends HttpServlet {
                 };
 
         rd.forward(wrapped, response);
+    }
+
+    private RequestParserHelper getRequestParserHelper(HttpServletRequest request) {
+        ServletContext servletContext = request.getServletContext();
+
+        WebApplicationContext webApplicationContext =
+                WebApplicationContextUtils.getRequiredWebApplicationContext(servletContext);
+
+        return webApplicationContext.getBean(RequestParserHelper.class);
     }
 
     @Override

--- a/src/main/resources/logback.xml
+++ b/src/main/resources/logback.xml
@@ -7,7 +7,7 @@
         class="ch.qos.logback.core.ConsoleAppender">
         <layout class="ch.qos.logback.classic.PatternLayout">
             <Pattern>
-                %d{ISO8601} %highlight(%-5level) [%t] %C{1.}: %msg%n%throwable
+                %d{ISO8601} %highlight(%-5level) [%t] %C{1}: %msg%n%throwable
             </Pattern>
         </layout>
     </appender>
@@ -17,7 +17,7 @@
         <file>${LOGS}/biblivre-logger.log</file>
         <encoder
             class="ch.qos.logback.classic.encoder.PatternLayoutEncoder">
-            <Pattern>%d %p %C{1.} [%t] %m%n</Pattern>
+            <Pattern>%d %p %C{1} [%t] %m%n</Pattern>
         </encoder>
  
         <rollingPolicy

--- a/src/test/java/biblivre/administration/report/ReportBOTest.java
+++ b/src/test/java/biblivre/administration/report/ReportBOTest.java
@@ -1,5 +1,7 @@
 package biblivre.administration.report;
 
+import static org.junit.Assert.assertNotNull;
+
 import biblivre.AbstractContainerDatabaseTest;
 import biblivre.administration.reports.ReportType;
 import biblivre.administration.reports.ReportsBO;
@@ -13,9 +15,6 @@ import biblivre.core.translations.Translations;
 import biblivre.core.translations.TranslationsMap;
 import biblivre.core.utils.Constants;
 import biblivre.core.utils.StringPool;
-
-import static org.junit.Assert.assertNotNull;
-
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;

--- a/src/test/java/biblivre/cataloging/RequestParserHelperImplTest.java
+++ b/src/test/java/biblivre/cataloging/RequestParserHelperImplTest.java
@@ -1,0 +1,31 @@
+package biblivre.cataloging;
+
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import biblivre.core.RequestParserHelper;
+import biblivre.core.RequestParserHelperImpl;
+import org.junit.jupiter.api.Test;
+
+public class RequestParserHelperImplTest {
+    @Test
+    public void testIsMustRedirectToSchema() {
+        RequestParserHelper requestParserHelper = new RequestParserHelperImpl();
+
+        assertFalse(requestParserHelper.isMustRedirectToSchema(""));
+    }
+
+    @Test
+    public void testParseController() {
+        RequestParserHelper requestParserHelper = new RequestParserHelperImpl();
+
+        assertEquals(requestParserHelper.parseController("", ""), "");
+
+        assertEquals(
+                requestParserHelper.parseController("DigitalMediaController", ""),
+                "DigitalMediaController");
+
+        assertNotEquals(requestParserHelper.parseController("foobar", ""), "foobar");
+    }
+}

